lib/graph/Module.js

```diff
@@ -27,6 +27,16 @@ function clone(obj) {
  * @alias @ui5/project/graph/Module
  */
 class Module {
+	/**
+	 * Regular expression to identify environment variables in strings.
+	 * Environment variables have to be prefixed with "env:", e.g. "env:Path".
+	 *
+	 * @private
+	 * @static
+	 * @readonly
+	 */
+	static _ENVIRONMENT_VARIABLE_REGEX = /env:\S+/g;
```

Comment:
I considered this aswell, but figured this would actually be expected behavior since environment variables behave like this in batch and powershell script string interpolation.

Consider a custom task that takes a configuration value in a specific format, e.g. something like
```yaml
configuration:
      example1: "key1=value1;key2=value2"
      example2: "value2"
```
With the current implementation, the user could inject their environment variables however they might like, e.g. something like
```yaml
configuration:
      example1: "key1=env:myvalue;key2=value2"
      example2: "env:myvalue"
```
This approach allows a more fine-grained configuration IMO. Now one might argue that the dev should not have implemented their custom configuration like this in the first place (and I agree, at least for this example a yaml array would have sufficed), but then again the goal is to move the responsibility away from the dev. 

I think it's very unlikely that a user might have a substring in a config value that coincidentally matches the environment variable regex given a sufficiently unique env var format, so the additional control and flexibility the user gets in return seems worth it.

I'll gladly change it if you want me to, though